snuba/query/parser/__init__.py

```diff
@@ -137,32 +138,34 @@ def build_selected_expressions(
         + build_selected_expressions(body.get("selected_columns", []))
     )
 
+    array_join_cols = set()
     arrayjoin = body.get("arrayjoin")
+    # TODO: Properly detect all array join columns in all clauses of the query.
+    # This is missing an arrayJoin in condition with an alias that is then
+    # used in the select.
     if arrayjoin:
+        array_join_cols.add(arrayjoin)
         array_join_expr: Optional[Expression] = parse_expression(
-            body["arrayjoin"], dataset.get_abstract_columnset()
+            body["arrayjoin"], dataset.get_abstract_columnset(), {arrayjoin}
         )
     else:
         array_join_expr = None
         for select_expr in select_clause:
             if isinstance(select_expr.expression, FunctionCall):
                 if select_expr.expression.function_name == "arrayJoin":
-                    if arrayjoin:
-                        raise ParsingException(
-                            "Only one arrayJoin(...) call is allowed in a query."
-                        )
-
                     parameters = select_expr.expression.parameters
                     if len(parameters) != 1 or not isinstance(parameters[0], Column):
                         raise ParsingException(
```

Comment:
If we allow multiple columns, then should this check go away too?

Reply:
Not enough. This PR now depends on https://github.com/getsentry/snuba/pull/1334 which takes care of multiple columns array joined in the query.

Reply:
No, I think every single arrayJoin call should have at most one parameter.

Reply:
@evanh actually I made a change here since Tony mentioned they need array join of literals